src/contracts/pods/EigenPod.sol

```diff
@@ -706,6 +718,28 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         return uint64(GENESIS_TIME + slot * SECONDS_PER_SLOT);
     }
 
+    /**
+     * @notice This function replicates the ssz hashing of a validator's pubkey, outlined below:
+     *  hh := ssz.NewHasher()
+     *  hh.PutBytes(validatorPubkey[:])
+     *  validatorPubkeyHash := hh.Hash()
+     *  hh.Reset()
+     */
+    function _hashPubkey(bytes memory validatorPubkey) internal pure returns (bytes32 pubkeyHash) {
+        require(validatorPubkey.length == 48, "Input should be 32 bytes in length");
+        bytes memory padding = new bytes(16);
+        bytes memory result = new bytes(64);
+
+        for (uint i = 0; i < validatorPubkey.length; i++) {
+            result[i] = validatorPubkey[i];
+        }
+        for (uint i = 0; i < padding.length; i++) {
+            result[i + validatorPubkey.length] = padding[i];
+        }
```

Comment:
nit: `uint` => `uint256`

Reply:
bigger issue here though -- is there a reason you aren't just using bytes.concat or abi.encodePacked to combine these bytes objects? looping byte-by-byte seems quite painful/inefficient

Reply:
documentation on `bytes.concat` with the version of solidity we're using: https://docs.soliditylang.org/en/v0.8.12/types.html#the-functions-bytes-concat-and-string-concat

Reply:
Fixed